src/python/WMCore/JobSplitting/JobFactory.py

```diff
@@ -27,11 +24,11 @@ class JobFactory(WMObject):
     def __init__(self,
                  package='WMCore.DataStructs',
                  subscription=None,
-                 generators=[],
+                 generators=None,
                  limit = 0):
         self.package = package
         self.subscription  = subscription
-        self.generators    = generators
+        self.generators    = generators if generators else []
```

Comment:
Well, the usual argument about not having empty lists or dicts as default arguments applies. We discussed this before, its just in general a really bad habit and you should never do it, even if in the specific case it doesn't hurt you.